### DISCUSSION TITLE
Animating viewBox with react-native-svg

### DISCUSSION THREAD

**Discussion by Andarius** [software-mansion/react-native-reanimated#2181] [Q&A] [answered]
Hey ! 
I'm trying to implement a zoom feature on a line chart using `react-native-svg`.
I'm not sure what the best option is, but I was looking into updating the `viewBox` prop to try emulating the zoom.
However, this code does not seem to have any effect:

```
const AnimatedSvg = Animated.createAnimatedComponent(Svg)

 const animatedSvgProps = useAnimatedProps<SvgProps>(() => {
        return {
            viewBox: `${translateX.value} 0 ${animatedWidth.value * scale.value} ${animatedHeight.value}`
        }
    })
<AnimatedSvg animatedProps={animatedSvgProps} />
```

Any ideas on why it's not updating here ?
It works if a re-render happens

**Comment by ericcAtGitHub** [comment]
https://github.com/software-mansion/react-native-reanimated/issues/3060#issuecomment-2559923940

**Comment by MatiPl01** [comment]
The right way to do this is to use the `Group` component from `react-native-svg`, wrap the chart within this component (likely the `Path` component you use to create a chart) and animate it using the `transform` property using the `useAnimatedProps` hook.

**Comment by Andarius** [comment, marked as answer]
For the record I implemented it like [this]( https://github.com/Andarius/rn-animations-playground/blob/master/src/screens/animations/LinechartScreen/Linechart/index.tsx) at the time : 

![scalableLinechart](https://github.com/user-attachments/assets/9c1321c8-d1f0-4d54-91dd-6657d45c8f6d)

**Reply by MatiPl01** [reply]
Thanks for your response. For this way of zooming chart, where you want the height to be unchanged and just expand plot in the x axis, your solution is better than just animating the `transform` property.
